### PR TITLE
feat: split the coordinator cron to single process

### PR DESCRIPTION
**Pull request by georgehao**
### Purpose or design rationale of this PR

*Describe your change. Make sure to answer these three questions: What does this PR do? Why does it do it? How does it do it?*

Because coordinator cron couldn't deploy duplicately. so need split the cron task as single process

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [ ] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: coordinator/cmd/cron/app/app.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	app.Name = "coordinator cron"
+	app.Usage = "The Scroll L2 Coordinator cron"
+	app.Version = version.Version
+	app.Flags = append(app.Flags, utils.CommonFlags...)
+	app.Flags = append(app.Flags, apiFlags...)
+	app.Before = func(ctx *cli.Context) error {
+		return utils.LogSetup(ctx)
+	}
+	// Register `coordinator-cron-test` app for integration-cron-test.
+	utils.RegisterSimulation(app, utils.CoordinatorCron)
```
Comment (mask-pp) [inline coordinator/cmd/cron/app/app.go]: `utils.CoordinatorCron` lead to ci failure.

File: common/utils/simulation.go [hunk 2]
```diff
@@ -25,7 +25,8 @@ var (
 	DBCliApp MockAppName = "db_cli-test"
 
 	// CoordinatorApp the name of mock coordinator app.
-	CoordinatorApp MockAppName = "coordinator-test"
+	CoordinatorApp     MockAppName = "coordinator-test"
+	CoordinatorCronApp MockAppName = "coordinator-cron-test"
```
Comment (mask-pp) [inline common/utils/simulation.go:30]:
```suggestion
	CoordinatorCronApp MockAppName = "coordinator_cron-test"
```

File: common/utils/simulation.go [hunk 3]
```diff
@@ -25,7 +25,8 @@ var (
 	DBCliApp MockAppName = "db_cli-test"
 
 	// CoordinatorApp the name of mock coordinator app.
-	CoordinatorApp MockAppName = "coordinator-test"
+	CoordinatorApp     MockAppName = "coordinator-test"
```
Comment (mask-pp) [inline common/utils/simulation.go]:
```suggestion
	CoordinatorApp     MockAppName = "coordinator_api-test"
```

File: coordinator/cmd/api/app/app_test.go [hunk 4]
```diff
@@ -10,7 +10,7 @@ import (
 )
 
 func TestRunCoordinator(t *testing.T) {
-	coordinator := cmd.NewCmd("coordinator-test", "--version")
+	coordinator := cmd.NewCmd("coordinator-api-test", "--version")
```
Comment (mask-pp) [inline coordinator/cmd/api/app/app_test.go:13]:
```suggestion
	coordinator := cmd.NewCmd(string(utils.CoordinatorAPIApp), "--version")
```

File: coordinator/cmd/cron/app/app_test.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	"fmt"
+	"testing"
+	"time"
+
+	"scroll-tech/common/cmd"
+	"scroll-tech/common/version"
+)
+
+func TestRunCoordinatorCron(t *testing.T) {
+	coordinator := cmd.NewCmd("coordinator-cron-test", "--version")
```
Comment (mask-pp) [inline coordinator/cmd/cron/app/app_test.go:13]:
```suggestion
	coordinator := cmd.NewCmd(string(utils.CoordinatorApp), "--version")
```

File: coordinator/cmd/cron/app/flags.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package app
+
+import "github.com/urfave/cli/v2"
+
+var (
+	apiFlags = []cli.Flag{
```
Comment (mask-pp) [inline coordinator/cmd/cron/app/flags.go]: `apiFlags` don't need this flags.

Comment (mask-pp) [reply coordinator/cmd/cron/app/flags.go]: ditto at api/app/flags.go

**Comment by mask-pp** [comment]
Please update readme doc in coordinator folder.

File: coordinator/cmd/api/app/app.go [hunk 7]
```diff
@@ -51,28 +50,23 @@ func action(ctx *cli.Context) error {
...
 	}()
 
+	registry := prometheus.DefaultRegisterer
+	observability.Server(ctx, db)
+
 	apiSrv := apiServer(ctx, cfg, db, registry)
 
 	log.Info(
-		"coordinator start successfully",
+		"Start coordinator successfully.",
```
Comment (mask-pp) [inline coordinator/cmd/api/app/app.go]:
```suggestion
		"Start coordinator api successfully.",
```

File: coordinator/cmd/cron/app/flags.go (hunk 6 above)
Comment (georgehao) [reply coordinator/cmd/cron/app/flags.go]: api/app/flags.go need

Comment (mask-pp) [reply coordinator/cmd/cron/app/flags.go]: > api/app/flags.go need

Only `httpPortFlag` is used, we can delete others.

Comment (georgehao) [reply coordinator/cmd/cron/app/flags.go]: have deleted [28e06c4](https://github.com/scroll-tech/scroll/pull/995/commits/28e06c476b1fce04a555b772993fc6272e4f6a85)

**Review by mask-pp — DISMISSED** [review]

File: coordinator/README.md [hunk 8]
```diff
@@ -44,13 +45,15 @@ The coordinator behavior can be configured using [`config.json`](config.json). C
...
 
 * Using manually specified ports and config.json:
 ```bash
-./build/bin/coordinator --config ./config.json --http --http.addr localhost --http.port 8390
+./build/bin/coordinator_api --config ./config.json --http --http.addr localhost --http.port 8390
+./build/bin/coordinator_cron --config ./config.json 
 ```
 
-* For other flags, refer to [`cmd/app/flags.go`](cmd/app/flags.go).
+* For other flcvgh bv cvbv ags, refer to [`cmd/api/app/flags.go`](cmd/api/app/flags.go).
```
Comment (0xmountaintop) [inline coordinator/README.md]: what's this?

**Review by 0xmountaintop — CHANGES_REQUESTED** [review]
`coordinator_cron` doesn't require `libzkp`

so we don't need to build `libzkp` in `coordinator_cron` docker
and can remove `libzkp` dependency for `coordinator_cron` (and no need `coordinator_cron_skip_libzkp`)

File: coordinator/README.md (hunk 8 above)
Comment (georgehao) [reply coordinator/README.md]: misinput

**Comment by georgehao** [comment]
> `coordinator_cron` doesn't require `libzkp`
> 
> so we don't need to build `libzkp` in `coordinator_cron` docker and can remove `libzkp` dependency for `coordinator_cron` (and no need `coordinator_cron_skip_libzkp`)

fixed

File: coordinator/Makefile [hunk 9]
```diff
@@ -53,7 +50,7 @@ clean: ## Empty out the bin folder
 	@rm -rf build/bin
 
 docker:
-	DOCKER_BUILDKIT=1 docker build -t scrolltech/coordinator-api:${IMAGE_VERSION} ${REPO_ROOT_DIR}/ -f ${REPO_ROOT_DIR}/build/dockerfiles/coordinator-api.Dockerfile
+	#DOCKER_BUILDKIT=1 docker build -t scrolltech/coordinator-api:${IMAGE_VERSION} ${REPO_ROOT_DIR}/ -f ${REPO_ROOT_DIR}/build/dockerfiles/coordinator-api.Dockerfile
```
Comment (0xmountaintop) [inline coordinator/Makefile]: forget to un-comment this line?

Comment (georgehao) [reply coordinator/Makefile]: test the coordinator_cron docker, forget to resume

**Review by 0xmountaintop — DISMISSED** [review]
LGTM

**Review by mask-pp — DISMISSED** [review]

**Comment by 0xmountaintop** [comment]
we'd better still bump the version for this PR, otherwise the tag will conflict

**Comment by georgehao** [comment]
> we'd better still bump the version for this PR, otherwise the tag will conflict

ok

**Pull request merged**